astropy/nddata/nddata.py

```diff
@@ -15,93 +15,95 @@
 
     uncertainty : any type, optional
         Uncertainty in the dataset.
-        Should have an attribute
-        ``uncertainty_type`` that defines what kind of uncertainty is stored,
-        for example ``"std"`` for standard deviation or
-        ``"var"`` for variance. If the uncertainty has no such attribute the
-        uncertainty is stored inside an `UnknownUncertainty` .
-        A metaclass defining such an interface is
-        `~astropy.nddata.NDUncertainty` but isn't mandatory.
+        Should have an attribute ``uncertainty_type`` that defines what kind of
+        uncertainty is stored, for example ``"std"`` for standard deviation or
+        ``"var"`` for variance. A metaclass defining such an interface is
+        `NDUncertainty` - but isn't mandatory. If the uncertainty has no such
+        attribute the uncertainty is stored as `UnknownUncertainty`.
         Defaults to ``None``.
 
     mask : any type, optional
-        Mask for the dataset.
-        Masks should follow the ``numpy`` convention that valid data points are
-        marked by ``False`` and invalid ones with ``True``.
+        Mask for the dataset. Masks should follow the ``numpy`` convention that
+        **valid** data points are marked by ``False`` and **invalid** ones with
+        ``True``.
         Defaults to ``None``.
 
     wcs : any type, optional
-        A world coordinate system (WCS) for the dataset.
+        World coordinate system (WCS) for the dataset.
         Default is ``None``.
 
     meta : `dict`-like object, optional
-        Meta information about the dataset. If no meta is provided an empty
-        `collections.OrderedDict` is created.
+        Additional meta informations about the dataset. If no meta is provided
+        an empty `collections.OrderedDict` is created.
+        Default is ``None``.
 
     unit : `~astropy.units.Unit`-like or str, optional
-        Unit for the dataset.
+        Unit for the dataset. Strings that can be converted to a
+        `~astropy.units.Unit` are allowed.
         Default is ``None``.
 
     copy : `bool`, optional
-        Save the attributes as copy or as reference. ``True`` copies every
-        attribute before saving it while ``False`` tries to save every
-        parameter as reference. Note however that it is not always possible to
-        save each input as reference.
+        Save the attributes as copy? ``True`` copies every attribute before
+        saving it while ``False`` tries to save every parameter as reference.
+        Note however that it is not always possible to save the input as
+        reference.
         Default is ``False``.
 
+        .. versionadded:: 1.2
+
     Raises
     ------
-    TypeError:
-        In case any parameter does not fulfill the classes restrictions.
+    TypeError
+        In case ``data`` or ``meta`` don't meet the restrictions.
+
+    Attributes
+    ----------
+    meta : `dict`-like
+        Additional meta informations about the dataset.
```

Comment:
Since I've changed the `meta` to use the descriptor `astropy.utils.metadata.MetaData` I couldn't find a way so this attribute is automatically documented. It seems like descriptors aren't recognized as attributes. The current manually included `meta` seems a bit out of place: http://astropy-mseifer.readthedocs.io/en/nddata_narrative_docs/api/astropy.nddata.NDData.html#nddata (in the attributes summary section)